ocp_resources/machine_config.py

```diff
@@ -1,5 +1,93 @@
+from typing import Any, Dict, List, Optional
 from ocp_resources.resource import Resource
 
 
 class MachineConfig(Resource):
+    """
+    https://docs.openshift.com/container-platform/4.15/rest_api/machine_apis/machineconfig-machineconfiguration-openshift-io-v1.html
+    """
+
     api_group = Resource.ApiGroup.MACHINECONFIGURATION_OPENSHIFT_IO
+
+    def __init__(
+        self,
+        os_image_url: Optional[str] = "",
+        config: Optional[Dict[str, Any]] = None,
+        kernel_arguments: Optional[List[str]] = None,
+        extensions: Optional[List[str]] = None,
+        fips: Optional[bool] = False,
```

Comment:
We do not set defaults, please set to None and use it only if the userr pass it.

Reply:
done